### PR TITLE
Add all the csi env

### PR DESCRIPTION
**Pull request by subhamkrai**
build: add all the csi image version in csv

add all the csi related images version in csv which
ocs-operator csv merger tool was changing.

build: These are auto-generated csv changes

these are auto-generated csv changes when run
`make gen-csv`.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: build/csv/ceph/rook-ceph-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -1863,12 +1863,22 @@ metadata:
   name: rook-ceph-operator.v4.15.0
   namespace: placeholder
   relatedImages:
```
Comment (b-ranto) [inline build/csv/ceph/rook-ceph-operator.clusterserviceversion.yaml:1865]: I believe this is wrong section for relatedImages, other operators have it in `spec`.

File: build/csv/csv-gen.sh [hunk 2]
```diff
@@ -54,6 +70,15 @@ function generate_csv() {
     sed -i "s/name: rook-ceph.v.*/name: rook-ceph-operator.v$CSV_VERSION/g" "$CSV_FILE_NAME"
     sed -i "s/version: 0.0.0/version: $CSV_VERSION/g" "$CSV_FILE_NAME"
 
+    # Update the csi version according to the downstream build env change
+    sed -i "s|quay.io/cephcsi/cephcsi:v3.10.2|$ROOK_CSI_CEPH_IMAGE|g" "$CSV_FILE_NAME"
```
Comment (Nikhil-Ladha) [inline build/csv/csv-gen.sh:74]: Let's use the LATEST_* varible here instead of the image so that it is easier to update images later on.

Comment (Nikhil-Ladha) [reply build/csv/csv-gen.sh:74]: And, all other places below

**Comment by b-ranto** [comment]
I have tested this locally. The CSI sidecars are properly being overloaded but none of the other images are, i.e. these images still ignore the env variables:
- ceph-csi
- csiaddons-sidecar
- rook-container

Weirdly, the ceph image disappeared from the relatedImages altogether. It is only available in the `alm-examples` section, now.

File: build/csv/csv-gen.sh (hunk 2 above)
Comment (subhamkrai) [reply build/csv/csv-gen.sh]: done

File: build/csv/ceph/rook-ceph-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (subhamkrai) [reply build/csv/ceph/rook-ceph-operator.clusterserviceversion.yaml]: updated

**Comment by b-ranto** [comment]
I have re-tested it and it works fine now. It turned out we were missing authetntication and that is why the three images above failed.

**Comment by subhamkrai** [comment]
> I have re-tested it and it works fine now. It turned out we were missing authetntication and that is why the three images above failed.

good to know it worked

**Comment by subhamkrai** [comment]
once #598 is merged will mark this ready for review

**Review by sp98 — APPROVED** [review]

**Review by Nikhil-Ladha — APPROVED** [review]
lgtm

**Comment by subhamkrai** [comment]
pushed with any modification to make commitlint happy

**Review by Nikhil-Ladha — APPROVED** [review]

**Review by b-ranto — APPROVED** [review]

**Pull request merged**